pytorch_lightning/trainer/trainer.py

```diff
@@ -1084,6 +1085,15 @@ def call_setup_hook(self, model: LightningModule) -> None:
         self.setup(model, stage=state)
         model.setup(stage=state)
 
+    def call_model_parallel_hook(self, model: LightningModule) -> None:
+        # Call model parallel hook if accelerator requests. In some cases
+        # we will not call the hook; the hook has initialized the sharded model for example.
+        if self.accelerator.call_model_parallel_setup_hook:
+            self.on_model_parallel_setup(model)
+            with self.accelerator.model_parallel_context():
+                model.on_model_parallel_setup()
```

Comment:
```suggestion
            with self.accelerator.model_parallel_context():
                self.on_model_parallel_setup(model)
                model.on_model_parallel_setup()
```

Reply:
I think the callback hook should also be within the context manager.

Reply:
Good catch!